addons/account_peppol/wizard/account_move_send.py

```diff
@@ -133,6 +133,16 @@ def action_send_and_print(self, force_synchronous=False, allow_fallback_pdf=Fals
 
         return super().action_send_and_print(force_synchronous=force_synchronous, allow_fallback_pdf=allow_fallback_pdf, **kwargs)
 
+    def _hook_if_errors(self, moves_data, from_cron=False, allow_fallback_pdf=False):
+        error_moves = self.env['account.move']
```

Comment:
```suggestion
       # Extends account
       moves_failed_ubl_generation = self.env['account.move']
```
Maybe you could rename `error_moves` so that it's clear that these are moves for which the UBL generation failed?

Reply:
hmm I don't think `blocking_error` only refers to ubl generation, I think it's any blocking error that occurs when generating a document in `_hook_invoice_document_before_pdf_report_render`, which includes ubl generation errors. I should actually probably just check for `error` in move_data, since that's what we check in the condition before calling the web service for peppol

Reply:
maybe something like this?

Reply:
I think it's great now clarity-wise, that's all I was asking for :smile: 

I think checking for `blocking_error` was actually better, because `_hook_if_errors` is also potentially called after the `_call_web_service` hooks at https://github.com/odoo/odoo/blob/622c018a654864b84b64d691baf9ab276363c45c/addons/account/wizard/account_move_send.py#L666

and so if you just check for errors, it might actually set the peppol state to skipped for invoices that had a peppol error which I don't think is what you want

whereas indeed as you said `blocking_error` is set only for documents where `_hook_invoice_document_before_pdf_report_render` fails